code/modules/projectiles/gun.dm

```diff
@@ -144,6 +151,10 @@
 		if(!can_trigger_gun(L))
 			return
 
+	if(safetyon == 1)
```

Comment:
shouldn't this be
if(safetyon || !can_shoot())  ?

That way it can be merged with the below if statement.